### DISCUSSION TITLE
Format "Script" tag in edge template

### DISCUSSION THREAD

**Discussion by feniljariwala82** [adonisjs/core#3066] [Help]
Does anyone have an idea about how to format the "Script" tag code in the edge template?

**Comment by McSneaky** [comment]
It depends on your editor. You can try to install "Edge" support, if that doesn't work, then just format it as HTML

Also depends if you use ESLint or Prettier or any other tool like that or just relying on editor autoformat

**Reply by feniljariwala82** [reply]
I use Vscode. And Prettier is not working in the edge template. If you have any idea about how to use Prettier with the edge template, please guide me.

**Reply by McSneaky** [reply]
No idea, never used Prettier :| 

But maybe there's some way to config Prettier to format it as HTML?

Here seems to be something https://stackoverflow.com/questions/57907645/prettier-not-formatting-html-files-in-vs-code

Perhaps can replace that `html` with `edge` in there, something like this:

```json
{
    "editor.formatOnSave": true,
    "editor.formatOnPaste": true,
    "editor.formatOnType": true,
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "[edge]": {
        "editor.defaultFormatter": "vscode.html-language-features"
    }
}
```

**Reply by feniljariwala82** [reply]
I am afraid to say this but "that didn't work!"

**Reply by grikomsn** [reply]
Solution is twofold: let vscode know that you can format `*.edge` files by adding this in your `settings.json` (either global or project):

```json
{
 "[edge]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "prettier.documentSelectors": ["**/*.edge"]
}
```

Then in your prettier config, add this to configure what parser to format `*.edge` files:

```json
{
  "overrides": [
    {
      "files": "*.edge",
      "options": {
        "parser": "html"
      }
    }
  ]
}
```

**Reply by thetutlage** [reply]
Wouldn't this mess up with the Edge tags or JS expressions you have written inside them?

**Reply by grikomsn** [reply]
Unfortunately that's true, and since @Julien-R44 is [not working on the prettier plugin](https://github.com/adonisjs/core/discussions/3989#discussioncomment-5794745), for now there's no proper solution for the formatting.

Personally would like to help out making one, need some pointers on prettier AST and how edge syntaxes work. 👀

**Reply by thetutlage** [reply]
Edge syntax is documented here https://github.com/edge-js/syntax. But sorry, not familiar with prettier API at all. 

Another way can be to port an existing Laravel Blade formatter (if that exists)

**Reply by grikomsn** [reply]
Thanks for the hint!

Technically it works using https://github.com/shufo/prettier-plugin-blade but it fails on `@set` with object params. Might need to partially modify the [formatter logic](https://github.com/shufo/blade-formatter/blob/0347239f4f9512c56c6e650c1ab3631ea3642bd8/src/formatter.ts) to allow curly braces since in php it uses square brackets.

```blade
 @set('$img', {
    'url': {
      'large': '/static/media/v2o/section-1/desktop/0.jpg',
      'medium': '/static/media/v2o/section-1/mobile/0.jpg'
    },
    'alt': '',
    'id': 'v2o-section-1-img-0'
})
```

```log
["ERROR" - 3:07:23 PM] Error formatting document.
["ERROR" - 3:07:23 PM] SyntaxError: Parse Error : syntax error, unexpected '{' on line 1
> 1 | <?php echo @set('$img', {
    |                        ^
  2 |                 'url': {
  3 |                   'large': '/static/media/v2o/section-1/desktop/0.jpg',
  4 |                   'medium': '/static/media/v2o/section-1/mobile/0.jpg'
Error: SyntaxError: Parse Error : syntax error, unexpected '{' on line 1
> 1 | <?php echo @set('$img', {
    |                        ^
  2 |                 'url': {
  3 |                   'large': '/static/media/v2o/section-1/desktop/0.jpg',
  4 |                   'medium': '/static/media/v2o/section-1/mobile/0.jpg'
```

Gonna tinker on the weekend and see if I can make it work.
